### PR TITLE
691 add tenant support to Binary on CW proxy

### PR DESCRIPTION
**Pull request by leite08**
Ref. metriport/metriport-internal#691

### Dependencies

https://github.com/metriport/metriport/pull/331

### Description

- Add tenant support to Binary on CW proxy
- Add logs when error while processing doc refs/binaries
- Add unit tests for CW FHIR proxy
- Add stricter `tsconfig` for specific path to make safer access to arrays

### Release Plan

- nothing special, asap

### CODE REVIEW

File: api/app/src/external/commonwell/cw-fhir-proxy.ts [hunk 1]
```diff
@@ -32,51 +38,72 @@ const updateQueryString = (path: string, params: string): string | undefined =>
...
-      proxyReqPathResolver: async function (req) {
-        console.log(`ORIGINAL HEADERS: `, JSON.stringify(req.headers));
-        console.log(`ORIGINAL URL: `, req.url);
-        const parts = req.url.split("?");
-        const path = parts[0];
-        const queryString = parts[1];
+export async function processDocReference(path: string, queryString?: string): Promise<MainType> {
+  if (!queryString) throw new BadRequestError(`Missing query string`);
+  const queryParams = new URLSearchParams(queryString);
+  const patienIdRaw = queryParams.get("patient.identifier")?.split("|") ?? [];
```
Comment (Goncharo) [inline api/app/src/external/commonwell/cw-fhir-proxy.ts:55]: nit: `patientIdRaw`

File: api/app/src/external/commonwell/document/document-download.ts [hunk 2]
```diff
@@ -19,6 +20,8 @@ export async function downloadDocument({
   location: string;
   stream: stream.Writable;
 }): Promise<void> {
+  const { log } = Util.out(`CW downloadDocument - M patient ${patientId}`);
```
Comment (Goncharo) [inline api/app/src/external/commonwell/document/document-download.ts:23]: nit: what does the `M` mean in this context? is this a typo?

**Review by Goncharo — APPROVED** [review]
couple tiny nits - otherwise LGTM :shipit:

File: api/app/src/external/commonwell/document/document-download.ts (hunk 2 above)
Comment (leite08) [reply api/app/src/external/commonwell/document/document-download.ts:23]: Metriport, so it doesn't eat too much space

**Pull request merged**